### DISCUSSION TITLE
I don't like how it ignores explicit constraints like "<4" when run

### DISCUSSION THREAD

**Discussion by Nixinova** [raineorshine/npm-check-updates#1407] [General]
This is very useful for keeping my packages up to date, but often I have to set a ceiling version number due to newer versions dropping support for Node versions. However this then makes this tool not useful at all as it ignores that constraint and tells me to update past that which I am incapable of doing.

I want the command to take that ceiling into account and only offer me upgrades that are below the ceiling I have set.
Upgrading past the ceiling point leaves me unable to see when there are genuine patch or minor upgrades etc that I would like to upgrade to.

Example current usage:
```
 binary-extensions       ^2.2.0  →  ^3.0.0
 commander           ^9.5.0 <10  →     ^12
 cross-fetch          ^3.1.8 <4  →      ^4
 isbinaryfile        ^4.0.10 <5  →      ^5
 typescript         ~5.0.4 <5.1  →    ^5.4
```

I thought `ncu --target semver` would do what I like but no.

Is there any usage that does do what I wish it to do?

Thanks

#1409

**Comment by raineorshine** [comment]
Hi, thanks for bringing this up.

I think what you've found is actually a bug with `--target semver`. It does not make any sense to exceed the specified range when trying to upgrade within semver. I will open an issue for this.

Fwiw, you can simply use `npm update` and npm will upgrade your installed packages (and lock file) to the latest version within the specified range. It is often not necessary to actually modify the versions in your `package.json` in this case.
